robot/packages/cloud_package_manager.go

```diff
@@ -364,6 +364,15 @@ func (m *cloudManager) downloadPackage(ctx context.Context, url string, p config
 		return err
 	}
 
+	// Delete legacy directory, silently fail if the cleanup fails
+	// This can be cleaned up after a few RDK releases (APP-4066)
+	if err := os.RemoveAll(p.LocalLegacyDataRootDirectory(m.packagesDir)); err != nil {
+		utils.UncheckedError(err)
+	}
+	if err := os.Remove(p.LocalLegacyDataRootDirectory(m.packagesDir)); err != nil {
+		utils.UncheckedError(err)
+	}
```

Comment:
```
// RemoveAll removes path and any children it contains.
// It removes everything it can but returns the first error
// it encounters. If the path does not exist, RemoveAll
// returns nil (no error).
// If there is an error, it will be of type *PathError.
func RemoveAll(path string) error {
```

[opt] this second remove is not required

Reply:
I initially tried this and it didn't work. But on 2nd thought I was having a lot of issues with the PR app image jobs yesterday and maybe i tested the wrong image. I'll try it again today.

Reply:
Turns out yup I just tested the wrong version of the image. So reverted that and will merge in now.